File: string_reverse.c
#include <string.h>

void main() {
    char str[] = "Hello, World!";
    int len = strlen(str);

    for (int i = 0; i < len / 2; i++) {
        char temp = str[i];
        str[i] = str[len - 1 - i];
        str[len - 1 - i] = temp;
    }
}

File: string_reverse.rs
fn main() {
    let mut str = String::from("Hello, World!");
    unsafe{
        let bytes = str.as_bytes_mut();
        let len = bytes.len();

        for i in 0..len / 2 {
            let temp = bytes[i];
            bytes[i] = bytes[len - 1 - i];
            bytes[len - 1 - i] = temp;
        }
    }
}
